url8db16c7d62_Arsaell_EdT_tgz-pJ8-DATA_MainJ8/src/Main.java

```diff
@@ -0,0 +1,59 @@
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+
+import javax.swing.JOptionPane;
+
+import DATA.DataStore;
+import DATA.Filler;
+import GUI.*;
+
+public class Main {
+
+	/**
+	 * Entrée du programme
+	 * @param args
+	 * @throws IOException 
+	 */
+	public static void main(String[] args) throws IOException	{
+		
+		// On cherche le datastore enregistré précédemment.
+		String workingDir = System.getProperty("user.dir");
+		
+		try {
+			BufferedReader br = new BufferedReader(new FileReader(workingDir + "/dataStoreLocation.loc"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *br*. Execution paths that do not contain closure statements were detected. Ensure *br* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).